src/components/structures/TimelinePanel.js

```diff
@@ -267,13 +286,15 @@ var TimelinePanel = React.createClass({
         // updates from pagination will happen when the paginate completes.
         if (toStartOfTimeline || !data || !data.liveEvent) return;
 
-        // even if we previously gave up forward-paginating, it's worth
-        // having another go now.
-        this.setState({canForwardPaginate: true});
-
         if (!this.refs.messagePanel) return;
 
-        if (!this.refs.messagePanel.getScrollState().stuckAtBottom) return;
+        if (!this.refs.messagePanel.getScrollState().stuckAtBottom) {
+            // we won't load this event now, because we don't want to push any
+            // events off the other end of the timeline. But we need to note
+            // that we can now paginate.
+            this.setState({canForwardPaginate: true});
```

Comment:
I need to return to this to fix https://github.com/vector-im/vector-web/issues/1427 anyway, so I'll go with it as-is for now.